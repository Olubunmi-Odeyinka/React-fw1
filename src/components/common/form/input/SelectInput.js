import React from 'react';
import PropTypes from 'prop-types';
import {Field} from 'formik';

const SelectInput = ({name, label, touched, error, options, focus, readOnly}) => {
  return (
    <div className="form-group">
      <label htmlFor={name}>{label}</label>
      <div className="field">
        {/* Note, value is set here rather than on the option - docs: https://facebook.github.io/react/docs/forms.html */}
        <Field
          component="select"
          name={name}
          className="form-control"
          readOnly={readOnly}
          autoFocus={focus}>
          <option value=""></option>
          {options && options.map((option) => {
            return <option key={option.value} value={option.value}>{option.text}</option>;
          })
          }
        </Field>
        {!readOnly && touched && error && <div className="alert alert-danger">{error}</div>}
      </div>
    </div>
  );
};

SelectInput.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  touched: PropTypes.bool,
  error: PropTypes.string,
  options: PropTypes.array,
  focus: PropTypes.bool,
  readOnly: PropTypes.bool
};

export default SelectInput;
