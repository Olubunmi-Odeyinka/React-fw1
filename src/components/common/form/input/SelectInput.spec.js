import React from 'react';
import { shallow } from 'enzyme';
import SelectInput from './SelectInput';

test('The Dropdown input component must render properly', ()=>{
  const props = {
    name: 'm',
    label: 'm',
    touched: false,
    error: '',
    focus: false,
    readOnly: false,
    options: []
  };

  const wrapper = shallow(<SelectInput {...props} />);
});

