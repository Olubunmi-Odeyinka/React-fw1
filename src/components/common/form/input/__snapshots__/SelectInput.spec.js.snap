// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The dropdown component input component must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelectInput
    error=""
    focus={true}
    label="Name"
    name="Name"
    options={Array []}
    readOnly={false}
    touched={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="Name"
>
          Name
</label>,
        <div
          className="field"
>
          <Field
                    autoFocus={true}
                    className="form-control"
                    component="select"
                    name="Name"
                    readOnly={false}
          >
                    <option
                              value=""
                    />
          </Field>
</div>,
      ],
      "className": "form-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Name",
          "htmlFor": "Name",
        },
        "ref": null,
        "rendered": "Name",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Field
              autoFocus={true}
              className="form-control"
              component="select"
              name="Name"
              readOnly={false}
>
              <option
                            value=""
              />
</Field>,
            false,
          ],
          "className": "field",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "autoFocus": true,
              "children": Array [
                <option
                  value=""
/>,
                Array [],
              ],
              "className": "form-control",
              "component": "select",
              "name": "Name",
              "readOnly": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "option",
              },
            ],
            "type": [Function],
          },
          false,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="Name"
>
            Name
</label>,
          <div
            className="field"
>
            <Field
                        autoFocus={true}
                        className="form-control"
                        component="select"
                        name="Name"
                        readOnly={false}
            >
                        <option
                                    value=""
                        />
            </Field>
</div>,
        ],
        "className": "form-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Name",
            "htmlFor": "Name",
          },
          "ref": null,
          "rendered": "Name",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Field
                autoFocus={true}
                className="form-control"
                component="select"
                name="Name"
                readOnly={false}
>
                <option
                                value=""
                />
</Field>,
              false,
            ],
            "className": "field",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "autoFocus": true,
                "children": Array [
                  <option
                    value=""
/>,
                  Array [],
                ],
                "className": "form-control",
                "component": "select",
                "name": "Name",
                "readOnly": false,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "option",
                },
              ],
              "type": [Function],
            },
            false,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
